pkg/syncer/storage/storage_pvc_process.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/logging"
+	"github.com/kcp-dev/kcp/pkg/syncer/shared"
+)
+
+const (
+	// DelayStatusSyncing instructs delaying the update of the PVC status back to KCP
+	DelayStatusSyncing = "internal.workload.kcp.dev/delaystatussyncing"
+	upSyncDiff         = "internal.workload.kcp.dev/upsyncdiff"
+)
+
+func (c *PersistentVolumeClaimController) process(ctx context.Context, key string) error {
+	logger := klog.FromContext(ctx)
+
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	logger = logger.WithValues(logging.NamespaceKey, namespace, logging.NameKey, name)
+
+	// Get the downstream PersistentVolumeClaim object for inspection
+	downstreamPVCObject, err := c.getDownstreamPersistentVolumeClaim(name, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get downstream PersistentVolumeClaim: %w", err)
+	}
+
+	downstreamPVC, ok := downstreamPVCObject.(*corev1.PersistentVolumeClaim)
+	if !ok {
+		return fmt.Errorf("failed to assert object to a PersistentVolumeClaim: %T", downstreamPVCObject)
+	}
+
+	logger.V(1).Info("processing downstream PersistentVolumeClaim")
+
+	_, ok = downstreamPVC.GetAnnotations()[DelayStatusSyncing]
+	// Is generation set to 0 when the object is created in KCP?
+	if !ok && downstreamPVC.Generation == 0 {
```

Comment:
We cannot probably depend on generation here.
We have to think a bit more about how we st it here and unset it in the PV controller, so that both controllers wouldn't fight.

Reply:
probably setting this whenever the PVC is not bound.